File: ios-native/IronCore/Models/LeaderboardEntry.swift
import Foundation
import FirebaseFirestore

/// Leaderboard entry at `leaderboard/{userId}`
struct LeaderboardEntry: Codable, Identifiable {
    @DocumentID var id: String?
    var userId: String
    var username: String
    var xp: Int
    var level: Int
    var league: String
    var avatarUrl: String
    var photo: String?
    var todayVolume: Double
    @ServerTimestamp var lastUpdated: Timestamp?
}

File: android-native/app/src/main/java/com/ironcore/fit/data/model/LeaderboardEntry.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: /leaderboard/{userId}
 * Denormalized for fast global queries — sorted by xp desc.
 */
data class LeaderboardEntry(
    val id: String = "",
    val username: String = "",
    val xp: Long = 0,
    val level: Int = 0,
    val league: String = "Iron Novice",
    val avatarUrl: String = "",
    val photo: String = "",
    val todayVolume: Long = 0,
    val rank: Int = 0,
    @ServerTimestamp val lastUpdated: Timestamp? = null
)
